sys/net80211/ieee80211_crypto_ccmp.c

```diff
@@ -377,8 +377,9 @@ ccmp_init_blocks(rijndael_ctx *ctx, struct ieee80211_frame *wh,
 	/* NB: aad[1] set below */
 	aad[2] = wh->i_fc[0] & 0x8f;	/* XXX magic #s */
 	aad[3] = wh->i_fc[1] & 0xc7;	/* XXX magic #s */
-	/* NB: we know 3 addresses are contiguous */
-	memcpy(aad + 4, wh->i_addr1, 3 * IEEE80211_ADDR_LEN);
+	memcpy(aad + 4, wh->i_addr1, IEEE80211_ADDR_LEN);
```

Comment:
Perhaps reword the commit log a bit to something like:

```
wlan_ccmp: Avoid memcpy that breaks with subobject bounds.

Copy each structure member explicitly instead of assuming they
are all contiguous.
```

In particular, this change doesn't really have anything to do with purecap, but only with subobject bounds (a purecap kernel without subobject bounds enabled wouldn't need this change)